docs/source/cookbook/libraries-with-smart-constructors.rst

```diff
@@ -0,0 +1,174 @@
+Libraries with smart constructors
+=================================
+
+Any type which uses a smart constructor (returning parsed result rather than
+throwing an exception) would require partial transformer rather than total
+transformer to convert.
+
+If there is no common interface which could be summoned as implicit for
+performing smart construction:
+
+.. code-block:: scala
+
+  // assuming Scala 3 or -Xsource:3 for fixed private constructors
```

Comment:
why this assumption is relevant?

Reply:
Without it `private` doesn't prevent generation of `Username.apply` nor `.copy()` so it is not a class with a smart constructor